### DISCUSSION TITLE
What is Sunrise/Sunset Duration (in mins) ?

### DISCUSSION THREAD

**Discussion by ze0ss** [t1m0thyj/WinDynamicDesktop#284] [Q&A] [answered]
What is it?
I just set it to 120 mins (which is the default, and was already there in the box)

I'm using the "Use specific sunrise and sunset times" option.

**Comment by t1m0thyj** [comment, marked as answer]
The way WDD's wallpaper engine works, it splits each 24-hour solar cycle into 4 segments: sunrise, day, sunset, and night.

If you input your location into the app, the sunrise and sunset durations it uses will be when the sun is between 6 degrees below and 6 degrees above the horizon. This duration usually happens to be around 2 hours, hence the default of 120 minutes if you select the manual schedule option.

**Reply by ze0ss** [reply]
But isn't a "horizon" just an arbitrary thing?
I mean, it exists, but it doesn't, at the same time, because it's imaginary, and is a concept meant to make humans understand sunrise/sunset and earth's rotation, right? (*middle-school flashbacks*)

But anyway, so what does duration actually mean, I'm so sorry I still don't get it properly.
120 min is default, so, how do I calculate how much it is, at some place "X"? Is there a formula to do it, better, any online calculator?
I searched for "Duration calculator" and all I could find is an astronomy.stackoverflow.com answer, which obviously wasn't what I wanted.

**Reply by t1m0thyj** [reply]
Yes, horizon is an arbitrary thing that changes around the world and based on time of year. The goal of the app is to follow the solar cycle as humans see/imagine it, similar to what Apple does with their dynamic wallpapers.

You can use the sun calculator here to find the durations for where you live: https://www.timeanddate.com/sun/
It shows the angle of the sun throughout the day, so you can use that to work out the approximately 120 minute duration when the sun is between +/- 6 deg.
The technical terms for this are "civil twilight" for 6 deg. below the horizon and "golden hour" for 6 deg. above.
